tests/integration/suite/daprd/workflow/backend/singular/activityactorstate.go

```diff
@@ -0,0 +1,343 @@
+	if runtime.GOOS == "windows" {
+		t.Skip("Skipping test on Windows due to SQLite limitations")
+	}
+
+	// Create a SQLite database
+	s.db = frameworksql.New(t,
+		frameworksql.WithActorStateStore(true),
+		frameworksql.WithTableName("workflow"),
+	)
+
+	s.handler = &httpServer{}
+
+	s.srv = prochttp.New(t, prochttp.WithHandler(s.handler.newhandler()))
+
+	s.place = placement.New(t,
+		placement.WithLogLevel("debug"), // TODO: rm
+	)
+
+	s.daprd = daprd.New(t,
+		daprd.WithLogLevel("debug"), // TODO: rm
+		daprd.WithResourceFiles(s.db.GetComponent(t)),
+		daprd.WithAppPort(s.srv.Port()),
+		daprd.WithPlacementAddresses("127.0.0.1:"+strconv.Itoa(s.place.Port())),
+	)
+
+	return []framework.Option{
+		framework.WithProcesses(s.db, s.daprd, s.place, s.srv),
+	}
+}
+
+func (s *activityactorstate) Run(t *testing.T, ctx context.Context) {
+	s.place.WaitUntilRunning(t, ctx)
+	s.daprd.WaitUntilRunning(t, ctx)
+
+	s.httpClient = fclient.HTTP(t)
+
+	conn, err := grpc.DialContext(ctx, //nolint:staticcheck
+		s.daprd.GRPCAddress(),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithBlock(), //nolint:staticcheck
+	)
+	require.NoError(t, err)
+	t.Cleanup(func() { require.NoError(t, conn.Close()) })
+
+	// Wait for actors to be ready
+	err = s.handler.waitForActorsReady(ctx)
+	require.NoError(t, err)
+
+	backendClient := client.NewTaskHubGrpcClient(conn, backend.DefaultLogger())
+
+	t.Run("check-activity-state", func(t *testing.T) {
+		r := task.NewTaskRegistry()
+		r.AddOrchestratorN("SingleActivity", func(ctx *task.OrchestrationContext) (any, error) {
+			var input string
+			if err := ctx.GetInput(&input); err != nil {
+				return nil, err
+			}
+			var output string
+			err := ctx.CallActivity("SayHello", task.WithActivityInput(input)).Await(&output)
+			return output, err
+		})
+
+		r.AddActivityN("SayHello", func(ctx task.ActivityContext) (any, error) {
+			var name string
+			if err := ctx.GetInput(&name); err != nil {
+				return nil, err
+			}
+			var r1, r2, r3, r4, r5, r6 []byte
+			queryCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+			defer cancel()
+			s.dbMutex.Lock()
+			rows, err := s.db.GetConnection(t).QueryContext(queryCtx, fmt.Sprintf("SELECT * FROM %s", s.db.GetTableName(t)))
+			s.dbMutex.Unlock()
+			require.NoError(t, err)
+			for rows.Next() {
+				err = rows.Scan(&r1, &r2, &r3, &r4, &r5, &r6)
+				if err != nil {
+					fmt.Printf("<<%v\n", err)
+					break
+				}
+				fmt.Printf(">>%s \n>>%s \n>>%s \n>>%s \n>>%s \n>>%s<<\n", r1, r2, r3, r4, r5, r6)
+
+				queryCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+				defer cancel()
+				s.dbMutex.Lock()
+				_, err := s.db.GetConnection(t).QueryContext(queryCtx, fmt.Sprintf("DELETE FROM %s", s.db.GetTableName(t)))
+				s.dbMutex.Unlock()
+				require.NoError(t, err)
+				fmt.Printf("cassie deelted from table")
+				break
+
+				// tried deleting all records
```

Comment:
I tried deleting everything from the table both from inside the activity func here and below as well, but still dont see the log about no activity state found, meaning it was always able to get the state in my test... Can this PR be accepted without a test since this is reverting a change in behavior that also had no test? 

Not sure if @JoshVanL you wanna take another stab at reproducing this issue? I thought my issue was due to the state being cached bc I see that is a field that can be set, but even when I manually changed it to not cache for the test runs I still was not seeing the no activity state found error.